### PR TITLE
Junit for Servlets2

### PR DESCRIPTION
**Pull request by subrat-dash-hcl**
Changes made to Servlets and Service classes. Added the Junit test classes

### CODE REVIEW

File: core/src/main/java/com/hcl/ecomm/core/services/ProductService.java [hunk 1]
```diff
@@ -13,11 +15,11 @@
 
 	public String getSearchCriteriaValue();
 
-	public JsonArray getAllProductDetails();
+	public JSONArray getAllProductDetails() throws JSONException;
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/services/ProductService.java:18]: @Subrat-Dash-416  Why this class changed?

File: core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java [hunk 2]
```diff
@@ -34,8 +34,8 @@
 
 	String scheme = "http";
 	String responseStream = null;
-	JsonArray productJsonArray = null;
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java:37]: @Subrat-Dash-416 16 Why this class changed?

File: core/src/main/java/com/hcl/ecomm/core/servlets/DeleteCartItemServlet.java [hunk 3]
```diff
@@ -68,9 +67,9 @@ protected void doPut(SlingHttpServletRequest request, SlingHttpServletResponse r
 			String payload = buffer.toString();
 			String customerToken = request.getHeader("CustomerToken");
 			if (StringUtils.isNotEmpty(payload)) {
-				JSONObject jsonPayload =  new JSONObject(payload);
+				JSONObject jsonPayload = new JSONObject(payload);
 				if (isValidItem(jsonPayload)) {
-					JSONObject deleteCartItemResponse = deleteCartItemService.deleteCartItem(jsonPayload.getString("cartId"), jsonPayload.getString("itemId"), customerToken);
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/servlets/DeleteCartItemServlet.java:73]: @Subrat-Dash-416  Why this class changed?

File: core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java [hunk 4]
```diff
@@ -37,19 +37,21 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
 			String customerToken = request.getHeader("CustomerToken");
 			String cartItems = null;
 			JSONObject responseObject = new JSONObject();
-			JSONObject responseStream = cartService.getCustomerCart(customerToken);
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java:40]: @Subrat-Dash-416 Why this class changed in junit?

File: core/src/main/java/com/hcl/ecomm/core/servlets/LoginServlet.java [hunk 5]
```diff
@@ -37,11 +37,15 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
 		LOG.info("loginService : " + loginService);
 		
 		
-		String token = loginService.getToken();
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/servlets/LoginServlet.java:40]: @Subrat-Dash-416  why this change required in junit?

File: core/src/main/java/com/hcl/ecomm/core/servlets/ProductDetailsServlet.java [hunk 6]
```diff
@@ -40,19 +40,19 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
 		String  sku = request.getParameter("sku");
         if(StringUtils.isNotEmpty(sku)) {
 		
-        JsonObject productResponse = productService.getProductDetail(sku);
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/servlets/ProductDetailsServlet.java:43]: @Subrat-Dash-416 Why this file change in junit ?

File: core/src/main/java/com/hcl/ecomm/core/servlets/ProductServlet.java [hunk 7]
```diff
@@ -33,20 +34,15 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
...
-			JsonArray responseStream = productService.getAllProductDetails();
-			JsonArray itemsarr= responseStream.getAsJsonArray();
-			LOG.info("itemsarr is {}", itemsarr.toString());
+			JSONArray responseStream = getAllProductDetails();
+			LOG.info("itemsarr is {}", responseStream.toString());
 			List<HashMap<String, String>> list = new ArrayList<HashMap<String, String>>();
-			JsonObject productObject = itemsarr.get(0).getAsJsonObject();
-			LOG.info("productobject is {}", productObject);
-			for(int i=0; i<itemsarr.size();i++){
+			for(int i=0; i<responseStream.length();i++){
```
Comment (rakeshkumar-verma) [inline core/src/main/java/com/hcl/ecomm/core/servlets/ProductServlet.java:40]: @Subrat-Dash-416  Why those changes required?

File: core/src/main/java/com/hcl/ecomm/core/services/ProductService.java (hunk 1 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/services/ProductService.java:18]: @rakeshkumar-verma  This change was made to resolve the google Gson error which we were facing .Discussed the same with you ,to follow the json.org approach.

File: core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java (hunk 2 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java:37]: same as above

File: core/src/main/java/com/hcl/ecomm/core/servlets/DeleteCartItemServlet.java (hunk 3 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/DeleteCartItemServlet.java:73]: @rakeshkumar-verma  As we need to surpass this method to return a json object which is required for the junit test class ,in order to cover this method call we have used spy by extending this line as method.

File: core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java (hunk 4 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/GetCustomerCartServlet.java:40]: same as above

File: core/src/main/java/com/hcl/ecomm/core/servlets/LoginServlet.java (hunk 5 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/LoginServlet.java:40]: same as above

File: core/src/main/java/com/hcl/ecomm/core/servlets/ProductDetailsServlet.java (hunk 6 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/ProductDetailsServlet.java:43]: same as above

File: core/src/main/java/com/hcl/ecomm/core/servlets/ProductServlet.java (hunk 7 above)
Comment (subrat-dash-hcl) [reply core/src/main/java/com/hcl/ecomm/core/servlets/ProductServlet.java:40]: @rakeshkumar-verma  bcoz we have changed from google json to normal json , so methods changes were required to match the same functionality .
